### DISCUSSION TITLE
Inwiefern kann man freetz-ng als Ersatz für Arm-SBCs Server nutzen?

### DISCUSSION THREAD

**Discussion by Miirpat-Gaalsien** [Freetz-NG/freetz-ng#645] [Allgemein]
Falls die Antwort "Wieso?" oder "Freetz ist dazu nicht gemacht" oder "Kauf dir ein Pi/NuC" ist, lies zumindest das untere.

Von der Perspektive von jemanden der Erfahrungen mit hosten auf Termux auf Android und SSH boxen ohne Root hat, währe es möglich ohne größeren Aufwand zur portierung Server Pakete via Debian/Alpine/Void/Gentoo/Docker zu nutzen? Folgende Methoden kenne ich, hat jemand welche davon probiert?:

- Namespaces (Docker/Podman/LXC/bwrap/usw), User namespaces
- Chroot
- Fakechroot (chroot via libc wrapper über LD_PRELOAD)
- Proot (chroot via ptrace)
- Gentoo prefix (Kompiliert geändern Root ordner in die Software)

Inwiefern währe ich begrenzt mit dem Packet Accelerator und geschlossenen kram? Gibt es größere Probleme externen Speicher zu nutzen?

Natürlich erwarte ich nicht die selbe Erfahrung wie auf einem Raspberry Pi, aber die sind längst etwas schwer zu bekommen, und die Alternativen sind mit der Dokumentation und Treiber/Kernel auch nicht so dufte, und bei den Kosten kommt noch der Strom hinzu, den ich durch DSL Router sowieso haben werde. Vielleicht werdet ihr mich überzeugen das es eine schlechte idee ist, aber für andere ist es eventuell gut genug.

**Comment by fda77** [comment]
Ich hab keine Ahnung um was es sich bei diesen Dingen handelt die du aufgezählt hast. Am besten probierst du es einfach mal aus. Wenn was nicht klappt wirst du es merken.
Wenn du unbedingt eine arm-cpu willst musst du dir halt eine Fritzbox aussuchen die eine hat

**Comment by pfichtner** [comment]
Ich hatte in der Richtung auch schon mal überlegt (weiß nicht mehr was der initale Auslöser dafür war). Würde das aber gar nicht auf ARM-CPUs beschränken wollen (ich war bei meinen Gedanken bei den TI CPUs). 

Von bereits existierenden Paketen/Images wird man gerade bei den letzteren genannten CPUs wenig bis gar nicht profitieren können (das war der eigentliche Grund für meinen Gedankengang). Die Frage wäre daher: Was wären dann die Gründe so einen Weg einschlagen zu wollen? Security? 
Die Alternative wäre ja: Die Pakete direkt als Freetz-Pakete einzubinden. 

Am Ende wird das was Deine Antritt war (Ersatz eines sonst zusätzlichen SoCs) wahrscheinlich am begrenzten RAM der FritzBoxen scheitern (ob mit oder ohne Isolierung/Containerisierung).  

Ich **vermute** der  Packet Accelerator wird da eher selten ein Problem darstellen, zumindest solange es sich nicht um sehr netzwerknahe Dinge handelt. Einen "normaler" Daemon sollte das halbwegs wurt sein.
